monai/transforms/utility/array.py

```diff
@@ -330,8 +330,10 @@ def __call__(self, img: NdarrayOrTensor, dtype: Optional[Union[DtypeLike, torch.
             TypeError: When ``img`` type is not in ``Union[numpy.ndarray, torch.Tensor]``.
 
         """
-        if not isinstance(img, (torch.Tensor, np.ndarray)):
-            raise TypeError(f"img must be one of (numpy.ndarray, torch.Tensor) but is {type(img).__name__}.")
+        if not isinstance(img, (torch.Tensor, np.ndarray)) and not (has_cp and isinstance(img, cp.ndarray)):
```

Comment:
Thanks, the underlying utility is quite robust, and will raise a valueerror if needed https://github.com/Project-MONAI/MONAI/blob/0c39281e31f8464503d3405526e85080e28c913f/monai/utils/type_conversion.py#L247

Perhaps we can remove this check in the transform?